django/forms/boundfield.py

```diff
@@ -231,11 +220,27 @@ def id_for_label(self):
         id_ = widget.attrs.get('id') or self.auto_id
         return widget.id_for_label(id_)
 
+    @property
```

Comment:
Agreed. I don't mind making a follow up PR with `cached_property` once this is resolved.